app/ante/utils/claim_rewards.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Evmos Foundation
+// This file is part of the Evmos Network packages.
+//
+// Evmos is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Evmos packages are distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Evmos packages. If not, see https://github.com/evmos/evmos/blob/main/LICENSE
+
+package utils
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// ClaimStakingRewardsIfNecessary checks if the given address has enough balance to cover the
+// given amount. If not, it attempts to claim enough staking rewards to cover the amount.
+func ClaimStakingRewardsIfNecessary(
+	ctx sdk.Context,
+	bankKeeper BankKeeper,
+	distributionKeeper DistributionKeeper,
+	stakingKeeper StakingKeeper,
+	addr sdk.AccAddress,
+	amount sdk.Coins,
+) error {
+	stakingDenom := stakingKeeper.BondDenom(ctx)
+	found, amountInStakingDenom := amount.Find(stakingDenom)
+	if !found {
+		// TODO: Should we return an error here? If the fees are not in the staking denom?
+		return nil
+	}
+
+	// TODO: Is this only giving the spendable balance or also the locked balance?
```

Comment:
total balance

Reply:
@fedekunze is there a way to check the "liquid" balance? Or do we have to query the locked tokens and then subtract those from the total balance?

Reply:
Do we know if things fail gracefully at least in case the is insufficient "liquid" amount available? 

In other words, if the account returned here is enough to pay the fees but not enough from it is liquid

Reply:
spendable balance is checked when you are using `SendCoins`, so it's already accounted for

Reply:
```suggestion
```